functions/postSurveyComplete.protected.ts

```diff
@@ -0,0 +1,142 @@
+  const handlerPath = Runtime.getFunctions()['helpers/hrmService'].path;
+  const buildDataObject = require(handlerPath)
+    .buildDataObject as BuildDataObject;
+
+  const taskAttributes = JSON.parse(surveyTask.attributes);
+
+  const data = buildDataObject(postSurveyConfigJson, memory);
+
+  const body: PostSurveyBody = {
+    contactTaskId: taskAttributes.contactTaskId,
+    taskId: surveyTask.sid,
+    data
+  };
+
+  await axios({
+    url: `${hrmBaseUrl}/postSurveys`,
+    method: 'POST',
+    data: JSON.stringify(body),
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': `Basic ${hrmStaticKey}`
+    },
+  });
+};
+
+export const handler: ServerlessFunctionSignature<EnvVars, Event> = async (
+  context: Context<EnvVars>,
+  event: Event,
+  callback: ServerlessCallback,
+) => {
+  console.log('-------- postSurveyComplete execution --------');
+  Object.entries(event).forEach(([k, v]) => console.log(k, JSON.stringify(v)));
+
+  try {
+    const memory = JSON.parse(event.Memory);
+    const { ServiceSid, ChannelSid } = memory.twilio.chat;
+
+    const client = context.getTwilioClient();
+
+    if (event.Channel === 'chat' && event.CurrentTask === 'complete_post_survey') {
+      const channel = await client.chat
+        .services(ServiceSid)
+        .channels(ChannelSid)
+        .fetch();
+
+      const channelAttributes = JSON.parse(channel.attributes);
+
+      if (channelAttributes.surveyTaskSid) {
+        // get the postSurvey definition
+        const serviceConfig = await client.flexApi.configuration.get().fetch();
+        const { definitionVersion, hrm_base_url, hrm_api_version } = serviceConfig.attributes;
+        const postSurveyConfigJson = Runtime.getAssets()[`/formDefinitions/${definitionVersion}/insights/postSurvey.json`];
+        const hrmBaseUrl = `${hrm_base_url}/${hrm_api_version}/accounts/${serviceConfig.accountSid}`;
+
+        // get the survey task
+        const surveyTask = await client.taskrouter
+          .workspaces(context.TWILIO_WORKSPACE_SID)
+          .tasks(channelAttributes.surveyTaskSid)
+          .fetch();
+
+        if (definitionVersion && postSurveyConfigJson && postSurveyConfigJson.open) {
+          const postSurveyConfigSpecs = JSON.parse(postSurveyConfigJson.open()) as OneToManyConfigSpecs;
+
+          // parallel execution to save survey collected data in insights and hrm
+          await Promise.all([
+            saveSurveyInInsights(postSurveyConfigSpecs, memory, surveyTask),
+            saveSurveyInHRM(postSurveyConfigSpecs, memory, surveyTask, hrmBaseUrl, context.HRM_STATIC_KEY),
+          ]);
+        } else {
+        // eslint-disable-next-line no-console
+          console.error('Missing definition for post survey. Not saving to insights.');
+        }
+
+        await surveyTask.update({ assignmentStatus: 'canceled' }); // can't complete a pending task only cancel it
```

Comment:
This comment is not clear.  Will this function only be called for "pending" tasks?